email-common/src/main/java/io/commercetools/sunrise/email/EmailSender.java

```diff
@@ -1,6 +1,82 @@
 package io.commercetools.sunrise.email;
 
+import javax.annotation.Nonnull;
+import javax.mail.internet.MimeMessage;
+import java.util.concurrent.*;
+
+/**
+ * A service for sending e-mails.
+ * <p>Instances of the e-mail service are created with a configuration such that they can later send messages
+ * without requiring further credentials or configuration of servers.
+ * <p>The service API is based on the Java Mail API. Note that implementations of this service might choose not to
+ * <em>send</em> e-mails using the (default implementation of the) Java Mail API but to merely accept messages created
+ * using the Java Mail API. You may want to consult the <a href="https://java.net/projects/javamail/pages/Home">samples,
+ * JavaDocs and specification of the Java Mail API</a>.
+ */
+@FunctionalInterface
 public interface EmailSender {
 
-    // TODO
+    /**
+     * Create a completion stage that asynchronously sends an e-mail using the configuration of this e-mail
+     * service. Before this method returns, the message author argument of this method is passed an empty
+     * {@link MimeMessage} that the {@link EmailSender} created. The message author shall prepare the message
+     * so it can be send. Sending will happen asynchronously, though.
+     * <p>
+     * Refer to the <a href="https://javamail.java.net/nonav/docs/api/">Java Mail JavaDoc</a> and the
+     * <a href="http://javamail.java.net/nonav/docs/JavaMail-1.5.pdf">Java Mail Specification</a> on how to configure
+     * instances of {@link MimeMessage}. The following code shows a simple example.
+     * <pre>{@code
+     * CompletionStage<String> completionStage = emailSender.send(msg -> {
```

Comment:
this is necessary because of the construction of the mail already requires the configuration?